bitcoin/src/crypto/key.rs

```diff
@@ -863,61 +890,84 @@ impl From<TweakedKeypair> for TweakedPublicKey {
     #[inline]
     fn from(pair: TweakedKeypair) -> Self { TweakedPublicKey::from_keypair(pair) }
 }
-/// A key-related error.
+/// Everyting perfect from here
+/// Error generated from WIF key format.
 #[derive(Debug, Clone, PartialEq, Eq)]
 #[non_exhaustive]
-pub enum Error {
+pub enum FromWifError {
     /// A base58 error.
     Base58(base58::Error),
-    /// A secp256k1 error.
-    Secp256k1(secp256k1::Error),
+    /// A secp256k1 error
+    Secp256k1(secp256k1::Error)
+}
+
+/// Error returned while generating key from slice.
+#[derive(Debug, Clone, PartialEq, Eq)]
+#[non_exhaustive]
+pub enum FromSliceError {
+    /// A base 58 error.
+    Base58(base58::Error),
     /// Invalid key prefix error.
     InvalidKeyPrefix(u8),
+    /// A Secp256k1 error,
+    Secp256k1(secp256k1::Error),
     /// Hex decoding error.
     Hex(hex::HexToArrayError),
     /// `PublicKey` hex should be 66 or 130 digits long.
     InvalidHexLength(usize),
```

Comment:
This should be just `InvalidLength` but I think `Secp256k1` already handles it so this should be deleted.

Reply:
Actually the `InvalidHexLength` is a variant because of this function [here](https://github.com/rust-bitcoin/rust-bitcoin/blob/6702f1a144801dc5e0c7779297311e76e4672933/bitcoin/src/crypto/key.rs#L239C14-L239C14). While generating `PublicKey` from hex this may occur.

Reply:
Oh, yes, `FromStr for PublicKey` must return yet another `ParsePubkeyError` that has the appropriate variants.